lucille-core/src/main/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnector.java

```diff
@@ -14,85 +15,99 @@
-  private List<String> otherJoinFields;
-  private List<Connection> connections = new ArrayList<Connection>();
+  private final String driver;
+  private final String connectionString;
+  private final String jdbcUser;
+  private final String jdbcPassword;
+  private final Integer fetchSize;
+  private final String preSql;
+  private final String sql;
+  private final String postSql;
+  private final String idField;
+  private final List<String> otherSQLs;
+  private final List<String> otherJoinFields;
+  private final List<Connection> connections = new ArrayList<>();
   // TODO: consider moving this down to the base connector class.
-  private ConnectorState state = null;
+  //  private ConnectorState state = null;
 
   // The constructor that takes the config.
   public DatabaseConnector(Config config) {
     super(config);
 
+    // required config
     driver = config.getString("driver");
     connectionString = config.getString("connectionString");
     jdbcUser = config.getString("jdbcUser");
     jdbcPassword = config.getString("jdbcPassword");
     sql = config.getString("sql");
     idField = config.getString("idField");
+
+    // optional config
+
     // For MYSQL this should be set to Integer.MIN_VALUE to avoid buffering the full resultset in memory.
     // The behavior of this parameter varies from driver to driver, often it defaults to 0.
-    if (config.hasPath("fetchSize"))
-      fetchSize = config.getInt("fetchSize");
-    if (config.hasPath("preSql"))
-      preSql = config.getString("preSql");
-    if (config.hasPath("postSql"))
-      postSql = config.getString("postSql");
+    fetchSize = config.hasPath("fetchSize") ? config.getInt("fetchSize") : null;
+    preSql = ConfigUtils.getOrDefault(config, "preSql", null);
+    postSql = ConfigUtils.getOrDefault(config, "postSql", null);
     if (config.hasPath("otherSQLs")) {
       otherSQLs = config.getStringList("otherSQLs");
       otherJoinFields = config.getStringList("otherJoinFields");
-    }  
+    } else {
+      otherSQLs = new ArrayList<>();
+      otherJoinFields = null;
+    }
   }
 
   // create a jdbc connection
   private Connection createConnection() throws ClassNotFoundException, SQLException {
-    Connection connection = null;
+    Connection connection;
     try {
       Class.forName(driver);
     } catch (ClassNotFoundException e) {
       log.error("Driver not found {} check classpath to make sure the jdbc driver jar file is there.", driver);
       setState(ConnectorState.ERROR);
-      throw(e);
+      throw (e);
     }
     try {
       connection = DriverManager.getConnection(connectionString, jdbcUser, jdbcPassword);
     } catch (SQLException e) {
       log.error("Unable to connect to database {} user:{}", connectionString, jdbcUser);
       setState(ConnectorState.ERROR);
-      throw(e);
+      throw (e);
     }
     connections.add(connection);
     return connection;
   }
 
   private void setState(ConnectorState newState) {

```

Comment:
This is dead code that is not used by anything in the codebase. It should be removed. If we want to add a concept of "state" to connectors, let's make a design ticket for that.

Reply:
Thx for creating the ticket.
Please remove the commented out code relating to setState -- having the ticket should be a good enough reminder and if we need to see history we can look back in git.
We should generally avoid having commented-out code in the codebase -- it creates clutter when reading code & it can't be maintained.

Reply:
done